src/python/nimbusml/tests/test_data_stream.py

```diff
@@ -30,6 +31,17 @@ def test_data_stream(self):
         assert repr(fi) == repr(fi2)
         os.remove(f.name)
 
+    def test_data_stream_path_object(self):
+        df = pandas.DataFrame(dict(a=[0, 1], b=[0.1, 0.2]))
+        with tempfile.NamedTemporaryFile(mode='w', delete=False) as f:
+            df.to_csv(f, sep=',', index=False)
+        
+        fi = FileDataStream.read_csv(Path(f.name), sep=',')
```

Comment:
The unit test is failing here. Looks like a similar change might also be required in `internal\utils\data_schema.py` (in `read_schema(*data, **options)`).

Here is the output from the build server:
``` console
_________________ TestDataStream.test_data_stream_path_object _________________
[gw0] win32 -- Python 3.7.3 D:\a\1\s\dependencies\Python3.7\python.exe

self = <nimbusml.tests.test_data_stream.TestDataStream testMethod=test_data_stream_path_object>

    def test_data_stream_path_object(self):
        df = pandas.DataFrame(dict(a=[0, 1], b=[0.1, 0.2]))
        with tempfile.NamedTemporaryFile(mode='w', delete=False) as f:
            df.to_csv(f, sep=',', index=False)
    
>       fi = FileDataStream.read_csv(Path(f.name), sep=',')

dependencies\Python3.7\lib\site-packages\nimbusml\tests\test_data_stream.py:39: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
dependencies\Python3.7\Lib\site-packages\nimbusml\internal\utils\utils.py:220: in wrapper
    params = func(*args, **kwargs)
dependencies\Python3.7\Lib\site-packages\nimbusml\internal\utils\data_stream.py:313: in read_csv
    filepath_or_buffer, nrows=nrows, **kwargs)
dependencies\Python3.7\Lib\site-packages\nimbusml\internal\utils\utils.py:220: in wrapper
    params = func(*args, **kwargs)
dependencies\Python3.7\Lib\site-packages\nimbusml\internal\utils\data_stream.py:347: in read_csv_pandas
    numeric_dtype=numeric_dtype, **kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

data = (WindowsPath('C:/Users/VSSADM~1/AppData/Local/Temp/tmphl_z3jyk'),)
options = {'collapse': False, 'numeric_dtype': None, 'sep': ','}, header = True
tool = 'pandas'
enumerate_blocks = <function DataSchema.read_schema.<locals>.enumerate_blocks at 0x0000018F9F573378>
cont = [WindowsPath('C:/Users/VSSADM~1/AppData/Local/Temp/tmphl_z3jyk')]

    @staticmethod
    def read_schema(*data, **options):
        """
        Infers the schema of a data view.
    
        :param data: features, labels, weights, groups
        :param collapse: (False by default), collapse columns for of the same type
          if it follows *read_csv* function. Use internal structure of a
          dataframe. If ``collapse* == 'all'``,
          the method collapses all columns not specified in parameter *names*.
        :param sep: string value of file seperation character (for example: ',')
        :param header: whether the data has a header row; defaults to True
        :param dtype: change dtype of specific columns; takes dictionary of column
          names mapped to desired dtype
        :param numeric_dtype: if not None, changes all numeric types into this type
        :param names: specify new names for columns; takes dictionary of column
          index mapped to desired name
        :param ind: first column index (in case DataFrame are concatenated)
        :param tool: `'pandas'` or `'nimbusml'`
        :return: schema as a string
    
        """
        if isinstance(options.get('dtype', None), dict) and \
                options.get('numeric_dtype', None):
            for k, v in options['dtype'].items():
                if DataColumn(name=k, type=v).is_numeric():
                    raise ValueError(
                        "dtype and numeric_dtype contradict. If "
                        "numeric_dtype is specified, dtype can only change "
                        "a column into a non numeric type ({0})".format(
                            options['dtype']))
    
        header = options.pop('header', True)
        ind = options.pop('ind', 0)
        tool = options.pop('tool', 'pandas')
    
        def enumerate_blocks(blocks):
            start = blocks[0]
            length = 1
            for b in blocks[1:]:
                if b == start + length:
                    length += 1
                else:
                    yield (start, length)
                    start = b
                    length = 1
            yield (start, length)
    
        def clean_name(col):
            if isinstance(col, tuple):
                # Multi-Index --> converted into a string.
                return '.'.join(clean_name(_) for _ in col)
            if isinstance(col, (str, text_type)):
                return col.strip().replace(" ", "").replace(':', '')
            elif isinstance(col, tuple):
                # multilevel index
                return col
            elif isinstance(col, six.integer_types):
                # reads a file with no header
                return "c%d" % col
            else:
                raise TypeError(
                    "col must be str or tuple not {0}".format(
                        type(col)))
    
        cont = [c for c in data if c is not None]
    
        if len(cont) == 1:
            X = cont[0]
    
            if isinstance(X, (ndarray, csr_matrix)):
                length = X.shape[1]
                sch = [
                    DataColumn(
                        name='Data',
                        pos=ind,
                        type=X.dtype,
                        length=length)]
            elif isinstance(X, Series):
                sch = [DataColumn(type=X.dtype, pos=ind, name=X.name)]
            elif isinstance(X, DataFrame):
                if options.get('collapse', False):
                    collapse = options['collapse']
                    if collapse not in (False, True, 'all'):
                        raise TypeError("collapse must be a boolean or 'all'")
                    if options.get('numeric_dtype'):
                        # We must change the type before but it should
                        # also affect the block.
                        dtype = options.get('numeric_dtype')
                        for c in X.columns:
                            td = DataColumn(name='_', type=X[c].dtype)
                            if td.is_numeric():
                                X[c] = X[c].astype(dtype)
                        # Merging blocks
                        if collapse is True:
                            names = list(X.columns)
                            dtype = dict(zip(names, X.dtypes))
                            st = StringIO()
                            X.to_csv(st, index=False, sep="\t", header=False)
                            st = StringIO(st.getvalue())
                            X = read_csv(
                                st,
                                sep="\t",
                                names=names,
                                dtype=dtype,
                                header=None,
                                error_bad_lines=False,
                                warn_bad_lines=False)
    
                    names = options.get('names', None)
                    if isinstance(names, dict):
                        names = set(_ for _ in names if isinstance(_, six.integer_types))
                    elif isinstance(names, list):
                        names = set(range(len(names)))
                    else:
                        names = set()
    
                    # breaking blocs
                    if collapse is True:
                        for n in names:
                            c = X.columns[n]
                            dt = X[c].dtype
                            X[c] = X[c].astype(str)
                            X[c] = X[c].astype(dt)
    
                        # merging columns
                        sch = []
                        cols = X.columns
                        res = []
                        pos = 0
                        for b in X._data.blocks:
                            row = eval(str(list(b._mgr_locs)))
                            for bl in enumerate_blocks(row):
                                n = cols[bl[0]]
                                if n.endswith("_0"):
                                    n = n[:-2]
                                sch.append(
                                    DataColumn(
                                        type=b.dtype,
                                        pos=bl[0] + ind,
                                        length=bl[1],
                                        name=n))
                            pos += b.shape[0]
                    elif collapse == 'all':
                        sch = []
                        keep = {}
                        for pos, (col, dtype) in enumerate(
                                zip(X.columns, X.dtypes)):
                            if pos in names:
                                sch.append(
                                    DataColumn(
                                        type=dtype,
                                        pos=pos,
                                        name=col))
                            else:
                                sdt = str(dtype)
                                if sdt not in keep:
                                    keep[sdt] = col, []
                                keep[sdt][1].append(pos + ind)
                        for k, v in sorted(keep.items()):
                            sch.append(
                                DataColumn(
                                    type=k, pos=v[1], length=len(
                                        v[1]), name=clean_name(
                                        v[0])))
                else:
                    sch = [DataColumn(type=X[c].dtype, pos=i + ind,
                                      name=clean_name(c), clean_name=True)
                           for i, c in enumerate(X.columns)]
            elif hasattr(X, 'read') or isinstance(X, str) or (
                    six.PY2 and isinstance(X, (str, text_type))):
                # A stream like StringIO.
                res = DataSchema.read_schema_file(
                    X, tool=tool, header=header, **options)
                return res
            elif isinstance(X, list):
                ser = DataFrame(X[:10] if len(X) > 10 else X)
                res = DataSchema.read_schema(ser, **options)
                return res
            else:
                raise TypeError(
                    "Unable to guess the schema for type '{0}'".format(
>                       type(X)))
E               TypeError: Unable to guess the schema for type '<class 'pathlib.WindowsPath'>'

dependencies\Python3.7\Lib\site-packages\nimbusml\internal\utils\data_schema.py:857: TypeError
```